tests/rptest/services/chaos/workloads/service_base.py

```diff
@@ -0,0 +1,281 @@
+#
+# Use of this software is governed by the Business Source License
+# included in the file licenses/BSL.md
+#
+# As of the Change Date specified in that file, in accordance with
+# the Business Source License, use of this software will be governed
+# by the Apache License, Version 2.0
+
+import requests
+import sys
+import os
+from enum import Enum
+from abc import ABC, abstractmethod
+import dataclasses
+
+from ducktape.services.service import Service
+from ducktape.tests.test import TestContext
+from ducktape.cluster.remoteaccount import RemoteCommandError
+from ducktape.utils.local_filesystem_utils import mkdir_p
+from ducktape.errors import TimeoutError
+
+from rptest.util import wait_until
+
+from ..types import NoProgressError
+
+
+class NodeState(str, Enum):
+    INITIALIZED = 'initialized'
+    STARTED = 'started'
+    STOPPED = 'stopped'
+
+
+@dataclasses.dataclass
+class WorkloadInfo:
+    succeeded_ops: int = 0
+    failed_ops: int = 0
+    timedout_ops: int = 0
+    is_active: bool = False
+
+
+class WorkloadServiceBase(ABC, Service):
+    @property
+    @abstractmethod
+    def java_module_name(self):
+        pass
+
+    @abstractmethod
+    def extra_config(self, node):
+        pass
+
+    @abstractmethod
+    def validate_consistency(self):
+        pass
+
+    @abstractmethod
+    def collect_stats(self):
+        pass
+
+    PERSISTENT_ROOT = "/var/lib/chaos_workload"
+    SYSTEM_LOG_PATH = os.path.join(PERSISTENT_ROOT, "system.log")
+    WORKLOAD_LOG_PATH = os.path.join(PERSISTENT_ROOT, "workload.log")
+
+    def __init__(self, ctx, brokers_str, num_nodes):
+        super().__init__(ctx, num_nodes=num_nodes)
+
+        self._remote_port = 8080
+        self._node_states = dict()
+        self._brokers_str = brokers_str
+
+        self.logs = {
+            "system": {
+                "path": self.SYSTEM_LOG_PATH,
+                "collect_default": True,
+            },
+            "workload": {
+                "path": self.WORKLOAD_LOG_PATH,
+                "collect_default": True,
+            }
+        }
+
+    def _remote_url(self, node, path):
+        return f"http://{node.account.hostname}:{self._remote_port}/{path}"
+
+    def _request(self, verb, node, path, timeout_sec=10, **kwargs):
+        kwargs["timeout"] = timeout_sec
+        url = self._remote_url(node, path)
+        self.logger.debug(f"dispatching {verb} {url}")
+        existing_exc = sys.exc_info()[1]
+        try:
+            r = requests.request(verb, url, **kwargs)
+        except requests.RequestException as e:
+            # Request exceptions (e.g. timeouts) come with a long context chain and
+            # one of those exceptions has the __suppress_context__ attribute set to True.
+            # This is not nice because the original context corresponding to existing_exc
+            # (e.g. a test error) will not not be printed. To avoid that, throw out
+            # all requests exceptions from the context chain.
```

Comment:
Thanks for investigation in the outdated discussion thread.

If we upgrade to urllib3>=2.0.0 we won't be affected anymore. Ducktape 0.11.4 does not depend on the "ugly" urrlib3 version. Is there any reason we use an old version of ducktape?

Reply:
Sounds like bumping it would be a separate piece of work :(

Ideally we should repair the exception chain: walk along it, remove the one thrown in the nasty python2-specific line and unset its parent's `__suppress_context__`. Something like this:
```
    except requests.RequestException as e:
        while e.__context__ is not None:
            if e.__context__.__class__ == TypeError and e.__suppress_context__ and e.__context__.args == ("HTTPConnection.getresponse() got an unexpected keyword argument 'buffering'",):
                print(e.__context__.args)
                e.__context__ = e.__context__.__context__
                e.__suppress_context__ = False
                break
            e = e.__context__
        raise
```
What do you think?

Reply:
> Ideally we should repair the exception chain

Makes sense, but
1) looks very brittle
2) the whole requests context chain doesn't look valuable to me, what I care about is that it is clear that a) there was a problem during the HTTP request and b) it was some kind of a timeout. Both of these are evident from the first exception in the chain and its stack trace.

So I think I would prefer the current code.